fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -4,53 +4,80 @@
 
 import com.yahoo.bard.webservice.data.config.names.TableName;
 import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
-import com.yahoo.bard.webservice.table.availability.Availability;
 import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
 import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
 
 import org.joda.time.DateTime;
-import org.joda.time.Interval;
 
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 /**
  * An interface describing a fact level physical table. It may be backed by a single fact table or multiple.
  */
 public interface PhysicalTable extends Table {
+    /**
+     * Getter for all the available intervals for the corresponding columns configured on the table.
+     *
+     * @return map of column to set of available intervals
+     */
+    Map<Column, SimplifiedIntervalList> getAllAvailableIntervals();
 
-    @Override
-    PhysicalTableSchema getSchema();
+    /**
+     * Return a view of the available intervals for this table given a data source constraint.
+     *
+     * @return The intervals that the table can report on, given the constraint
+     */
+    default SimplifiedIntervalList getAvailableIntervals() {
+        return getAllAvailableIntervals().values()
+                .stream()
+                .reduce(SimplifiedIntervalList::union)
+                .orElse(SimplifiedIntervalList.empty());
+    }
 
     /**
-     * Get the name of the current table.
+     * Return a view of the available intervals for this table given a constraint.
      *
-     * @return name of the table as TableName
+     * @param constraint  The constraint which limits available intervals
+     *
+     * @return The intervals that the table can report on
```

Comment:
seems like the comment is flipped with the above no argument version of this method.